apps/src/pythonlab/pyodideWorkerManager.ts

```diff
@@ -53,22 +54,30 @@ const setUpPyodideWorker = () => {
         break;
       case 'system_error':
         getStore().dispatch(appendSystemError(message));
-        MetricsReporter.logError({
-          type: 'PythonLabSystemCodeError',
-          message,
-        });
+        Lab2Registry.getInstance()
+          .getMetricsReporter()
+          .logError('Python Lab System Code Error', undefined, {message});
         break;
       case 'internal_error':
-        MetricsReporter.logError({
-          type: 'PythonLabInternalError',
-          message,
-        });
+        Lab2Registry.getInstance()
+          .getMetricsReporter()
+          .logError('Python Lab Internal Error', undefined, {message});
         break;
       case 'loading_pyodide':
         getStore().dispatch(setLoadedCodeEnvironment(false));
+        loadStartTime = Date.now();
```

Comment:
Not super familiar with this code, but is this guaranteed to always get a `loading` event before a `loaded` event? Asking because I think you could alternatively track load time in the web worker and pass it back as part of the event data?

Reply:
Oh good call. We should always get the loading event first, but your idea is better, will update.